llama2.mojo

```diff
@@ -431,21 +429,19 @@ struct TransformerWeights:
 
 
 fn read_file(file_name: String, inout buf: FileBuf) raises:
-    let _os = Python.import_module("os")
-    let ff_size = _os.path.getsize(file_name)
-    let cp_size = string.atol(ff_size.to_string())
+    let fd = open(file_name, "r")
+    let data = fd.read()
+    fd.close()
+
+    let cp_size = data._buffer.size
     let cp_buf: BufferPtrType = BufferPtrType.alloc(cp_size)
-    # set window buffer to read binary data from file
-    let f = File(file_name)
-    var reader = BufReader[4096](f ^)
-    var bytes_read = 1
-    var offset = 0
-
-    while bytes_read > 0:
-        let buf = Buffer[4096, DType.uint8](cp_buf.offset(offset))
-        bytes_read = reader.read(buf)
-        offset += bytes_read
-    reader.do_nothing()  # keeps lifetimes working
+
+    let data_ptr = data._as_ptr().bitcast[DType.uint8]()
+    
+    for i in range(cp_size):
+        cp_buf.store(i,data_ptr.load(i))
```

Comment:
I think here it reads data twice into memory. Could you try to reuse the data buffer, need to set cp_buf buffer to this data._buffer somehow and make sure `data` var won't destruct after exiting `fn`

Reply:
Yes it would be possible but i don't know if it is memory safe:
```python
var fd = open("stories15M.bin","r")
var data = fd.read()
fd.close()

let cp_buf = data._as_ptr().bitcast[DType.uint8]()
data._buffer.data = data._buffer.data.get_null().alloc(1)
data._buffer.size = 1
data._buffer.capacity = 1
_=data
```
[Dynamic vector](https://docs.modular.com/mojo/stdlib/utils/vector.html#dynamicvector)
it might call ```__del__```on its elements in the future

Reply:
@rd4com I think it's worthwhile to ask this question on Discord.. I'm a bit concerned that llama2.mojo users already have OOM errors, and now if we introduce this change we could have more issues like that because llama2 will require 2x times more memory (1) to read file , (2) to fill the structs

I filed a few Feature Requests to mojo , let's see maybe they can help with this soon .. 

And also, let's ask on Discord about your latest code? Is it safe or not

Reply:
sure, that is the wise thing to do! you ask or i do ?

Reply:
@rd4com you got this initiative, so feel free to be the owner :)